internal/cmdparser/test.go

```diff
@@ -45,16 +54,33 @@ func TestCmd[T PtrAsReceiverWrapper[pointerType], pointerType any](args ...strin
 
 func testCmd[T PtrAsReceiverWrapper[pointerType], pointerType any](args ...string) error {
 	c := New[T](dependency.Options{
-		Output: output.New(output.Options{LoggingLevel: verbosity.Trace}),
+		Output: output.New(output.Options{
+			LoggingLevel: verbosity.Trace}),
 	})
 	c.DefineCommand()
 	if len(args) > 1 {
-		panic("Only provide one string of args, they will be split on space")
+		panic("Only provide one string of args, they will be split on space/quoted values (with spaces)")
 	} else if len(args) == 1 {
-		c.SetArgsForUnitTesting(strings.Split(args[0], " "))
+		c.SetArgsForUnitTesting(splitStringIntoArgsSlice(args[0]))
 	} else {
 		c.SetArgsForUnitTesting([]string{})
 	}
 	err := c.Command().Execute()
 	return err
```

Comment:
>urn err [](http://example.com/codeflow?start=4&length=7)

should this return the output as well so a test can verify the command did more than not crash?

Reply:
Good feedback, it should (I was concentrating more on getting high cover coverage, but you are right, we need better verification.  

I think the right abstraction here, is a function with an extra parameter, something TestCmd(args, regExToVerify).

Addedissue for this:

[https://github.com/microsoft/go-sqlcmd/issues/215](https://github.com/microsoft/go-sqlcmd/issues/215)